cirq-core/cirq/ops/pauli_string_phasor.py

```diff
@@ -117,15 +146,131 @@ def merged_with(self, op: 'PauliStringPhasor') -> 'PauliStringPhasor':
+            dense_pauli_string: The DensePauliString defining the positive and
+                negative eigenspaces that will be independently phased.
+            exponent_neg: How much to phase vectors in the negative eigenspace,
+                in the form of the t in (-1)**t = exp(i pi t).
+            exponent_pos: How much to phase vectors in the positive eigenspace,
+                in the form of the t in (-1)**t = exp(i pi t).
+
+        Raises:
+            ValueError: If coefficient is not 1 or -1.
+        """
+        if dense_pauli_string.coefficient == -1:
+            dense_pauli_string = -dense_pauli_string
+            exponent_pos, exponent_neg = exponent_neg, exponent_pos
+
+        if dense_pauli_string.coefficient != 1:
+            raise ValueError(
+                "Given PauliString doesn't have +1 and -1 eigenvalues. "
+                "pauli_string.coefficient must be 1 or -1."
+            )
+
+        self._dense_pauli_string = dense_pauli_string
+        self._exponent_neg = value.canonicalize_half_turns(exponent_neg)
+        self._exponent_pos = value.canonicalize_half_turns(exponent_pos)
+
+    @property
+    def exponent_relative(self) -> Union[int, float, sympy.Basic]:
+        return value.canonicalize_half_turns(self.exponent_neg - self.exponent_pos)
+
+    @property
+    def exponent_neg(self):
+        return self._exponent_neg
+
+    @property
+    def exponent_pos(self):
+        return self._exponent_pos
+
+    @property
+    def dense_pauli_string(self):
+        return self._dense_pauli_string
+
+    def _value_equality_values_(self):
+        return (
+            self.dense_pauli_string,
+            self.exponent_neg,
+            self.exponent_pos,
+        )
+
+    def equal_up_to_global_phase(self, other):
+        if isinstance(other, PauliStringPhasorGate):
+            rel1 = self.exponent_relative
+            rel2 = other.exponent_relative
+            return rel1 == rel2 and self.dense_pauli_string == other.dense_pauli_string
+        return False
+
+    def __pow__(self, exponent: Union[float, sympy.Symbol]) -> 'PauliStringPhasorGate':
+        pn = protocols.mul(self.exponent_neg, exponent, None)
+        pp = protocols.mul(self.exponent_pos, exponent, None)
+        if pn is None or pp is None:
+            return NotImplemented
+        return PauliStringPhasorGate(self.dense_pauli_string, exponent_neg=pn, exponent_pos=pp)
+
     def _has_unitary_(self):
         return not self._is_parameterized_()
 
-    def _decompose_(self) -> 'cirq.OP_TREE':
-        if len(self.pauli_string) <= 0:
+    def _to_z_basis_ops(self, qubits: Sequence['cirq.Qid']) -> Iterator[raw_types.Operation]:
+        """Returns operations to convert the qubits to the computational basis."""
+        for i in range(len(self.dense_pauli_string)):
+            yield clifford_gate.SingleQubitCliffordGate.from_single_map(
+                {self.dense_pauli_string[i]: (pauli_gates.Z, False)}
+            )(qubits[i])
```

Comment:
Use `self.dense_pauli_string.on(*qubits).to_z_basis_ops()` instead of re-writing the method here.